include/mirage/transpiler/runtime/kernel/element_unary.h

```diff
@@ -9,19 +9,24 @@
 
 namespace kn {
 
-enum class ElementUnaryOpType { EXP };
+enum class ElementUnaryOpType { EXP, SILU, SQUARE, SQRT };
 
 template <typename T, ElementUnaryOpType OP>
 static __device__ __forceinline__ T perform_element_unary_op(T a) {
+  if constexpr (!(std::is_same_v<T, cutlass::half_t> ||
+                  std::is_same_v<T, __half>)) {
+    assert(0 && "unsupport datatype in kn elementunary");
+  }
   if constexpr (OP == ElementUnaryOpType::EXP) {
-    if constexpr (std::is_same_v<T, cutlass::half_t> ||
-                  std::is_same_v<T, __half>) {
-      return (T)expf((float)a);
-    } else {
-      assert(0);
-    }
+    return (T)expf((float)a);
+  } else if (OP == ElementUnaryOpType::SILU) {
```

Comment:
Checking operator type at run time is very expensive. Can we add `constexpr` as a quantifier for all the following if conditions?

Reply:
done!